File: app/src/main/java/com/team/eddie/uber_alles/utils/LoginServices.java
package com.team.eddie.uber_alles.utils;

import retrofit2.Call;
import retrofit2.http.Field;
import retrofit2.http.FormUrlEncoded;
import retrofit2.http.POST;

public interface LoginServices {

    /**
     * Login user
     *
     * @param username
     * @param password
     */
    @FormUrlEncoded
    @POST("login")
    Call<Void> userLogin(@Field("username") String username, @Field("password") String password);

    /**
     * Logout
     */
    @POST("logout")
    Call<Void> logout();
}

File: app/src/main/java/com/team/eddie/uber_alles/utils/LoginServices.kt
package com.team.eddie.uber_alles.utils

import io.reactivex.Single
import retrofit2.Call
import retrofit2.http.Body
import retrofit2.http.POST
import retrofit2.http.QueryMap


interface LoginServices {

    /**
     * Login user
     *
     * @param username
     * @param password
     */
    @POST("user/login")
    fun userLogin(@QueryMap params: Map<String, String>): Call<Void>

    @POST("login")
    fun login(@Body user: User): Single<User>

    /**
     * Logout
     */
    @POST("logout")
    fun logout(): Call<Void>
}